### PR TITLE
Routes for going directly to Become a Provider

### PR DESCRIPTION
**Pull request by shannonwells**
## Goal

The goal of this PR is to lay groundwork for the new Provider registration flow by setting up routes for going directly to the become-a-provider route.  

*Edit* Filed issue #194 for looking at a non-hack fix for broken tests.

Related to #184 

## Discussion
Features:
* The URL may include the name of the network to connect to. If so, it will try to connect to the given network.
* The network selection dropdown goes away (on Provider Login and Become a Provider pages) when a network is selected.  A 'Change network' link is provided which resets the component.
* The network part of the URL is optional.

So the following routes all work, and they are case-insensitive.
1. `/become-a-provider`:  which displays the network dropdown as before
2. `/become-a-provider/mainnet` : connects to mainnet immediately from the page
3. `/become-a-provider/testnet`: connects to Paseo testnet
4. `/become-a-provider/localhost` : tries to connect to localhost:9944
5. `become-a-provider/MaInNet`, etc.
6. `/mainnet`, `/testnet`, `/localhost` will each connect to the main page, select the appropriate network from the dropdown as above, and try to connect to it.
7. Custom network connections work as before.

## Checklist
- [x] routes for become-a-provider
- [x] routes for root (provider login)
- [x] PR Self-Review and Commenting
- [x] Tests added

## How To Test the Changes
Check out the branch and run the playwright tests.
```shell
git fetch && git co feat/new-registration-path
npx playwright test
```
Then you can play around with the new feature via `npm run dev`.

You should be able to click back and forth between Provider Login and Become a Provider pages, selecting things, then clicking Back/Cancel, and the state of the selections and connections should get reset.

You should be able to go to the URLs listed above and either connect automatically, or select from dropdown and connect, then click 'Change network' and the state should also be reset.

### CODE REVIEW

File: src/lib/stores/networksStore.ts [hunk 1]
```diff
@@ -12,24 +12,38 @@ export enum NetworkType {
...
   id: NetworkType;
-  name: string;
+  displayName: string;
+  pathName: string;
   endpoint?: string;
   genesisHash?: string;
 }
 
-export const allNetworks = writable<NetworkInfo[]>([
+export const defaultNetworks: NetworkInfo[] = [
```
Comment (shannonwells) [inline src/lib/stores/networksStore.ts:21]: this refactor was just to support converting from the path to the NetworkInfo.

File: src/routes/become-a-provider/[[network=networks]]/+layout.svelte [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<script>
```
Comment (shannonwells) [inline src/routes/become-a-provider/[[network=networks]]/+layout.svelte:1]: note the path to this file, it uses the `match` function above to figure out if the route is valid, and then populates $page.params accordingly with a value for `network`, or not.

File: playwright.config.ts [hunk 3]
```diff
@@ -75,6 +75,6 @@ export default defineConfig({
   webServer: {
     command: 'npm run build && npm run preview',
     port: 4173,
-    reuseExistingServer: !process.env.CI,
+    reuseExistingServer: false,
```
Comment (shannonwells) [inline playwright.config.ts:78]: Having this set to true breaks tests.

File: src/components/BecomeAProvider.svelte [hunk 4]
```diff
@@ -16,7 +16,7 @@
       <SelectNetworkAndAccount
         bind:newUser={$user}
         accounts={$nonProviderAccountsStore}
-        accountSelectorTitle="Account Id"
+        accountSelectorTitle="Select an Account Id"
```
Comment (shannonwells) [inline src/components/BecomeAProvider.svelte:19]: to match the Become a Provider prompt

File: src/components/LoginForm.svelte [hunk 5]
```diff
@@ -32,12 +32,12 @@
...
   accounts={$providerAccountsStore}
-  accountSelectorTitle="Select an Account Id"
+  accountSelectorTitle="Select a Provider Account Id"
   accountSelectorPlaceholder="Select a Provider Account Id"
-  noAccountsFoundErrorMsg="No Provider Account Ids found.  Become a Provider?"
+  noAccountsFoundErrorMsg="No Provider Account Ids found. To become a Provider, see below."
 />
 <div class="flex justify-between align-bottom">
-  <Button id="connect-button" title="Connect" disabled={!canConnect} action={connect} />
+  <Button id="connect-button" title="Connect To Account" disabled={!canConnect} action={connect} />
```
Comment (shannonwells) [inline src/components/LoginForm.svelte:40]: "Connect" on the button was confusing due to the other text if you connected to the endpoint.  So make the button more clear what you're connecting to.

File: src/params/networks.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const match = (path: string): boolean => {
+  return (/mainnet|testnet|localhost/i).test(path);
+}
```
Comment (shannonwells) [inline src/params/networks.ts:3]: See defaultAccounts `pathname`

**Review by JoeCap08055 — COMMENTED** [review]
Couple of things:

1. Playwright tests are passing, but other tests appear to be broken (both when I run locally & in pipeline)
2. The optional path component for the network is interesting; I'm wondering if you considered a more traditional URL query parameter approach, ie `?node=<mainnet|testnet|URL>` (similar to the PolkadotJS UI)? Would seemingly allow for custom networks (or even just specific nodes) right in the link, as opposed to just mainnet/testnet/localhost _specific_ node. Just my $0.02.
:point_up: non-blocking, but tests should def. be fixed

**Comment by shannonwells** [comment]
> The optional path component for the network is interesting; I'm wondering if you considered 

I did, but reading through the docs it didn't seem to be the Svelte Way, and once I already had it implemented, I realized how to do it the other way.  ¯\_(ツ)_/¯

I could be persuaded to redo it and use query params instead.

**Comment by JoeCap08055** [comment]
Hmmm, maybe out of scope, but... should the page maybe retain your current network choice/connection when you navigate back to Home? As of now it does not appear to.

Example:
* Navigate to http://localhost:<port>/localhost (or just select a network from the dropdown from the home page)
* Click "Become a Provider"
* Navigate back to "Home" using the link on the page

Expected results: Still connected to "localhost" node
Actual results: Required to select a network connection again

**Comment by shannonwells** [comment]
> should the page maybe retain your current network choice/connection when you navigate back to Home? As of now it does not appear to.

IMO, no.  If you navigate back to home then you've changed your mind about what you want to do.  We can't predict what the user intent was and it's harmless to make them do another dropdown selection and reconnect.

**Pull request merged**